pkg/component/component.go

```diff
@@ -211,6 +211,32 @@ func ListAllClusterComponents(client kclient.ClientInterface, namespace string)
 	return components, nil
 }
 
+func ListAllComponents(client kclient.ClientInterface, namespace string, devObj parser.DevfileObj) ([]api.ComponentAbstract, string, error) {
+	devfileComponents, err := ListAllClusterComponents(client, namespace)
+	if err != nil {
+		return nil, "", err
+	}
+
+	var localComponent api.ComponentAbstract
+	if devObj.Data != nil {
+		localComponent = api.ComponentAbstract{
+			Name:      devObj.Data.GetMetadata().Name,
+			ManagedBy: "",
+			RunningIn: []api.RunningMode{},
+			Type:      GetComponentTypeFromDevfileMetadata(devObj.Data.GetMetadata()),
+		}
+	}
+
+	componentInDevfile := ""
+	if localComponent.Name != "" {
+		if !Contains(localComponent, devfileComponents) {
+			devfileComponents = append(devfileComponents, localComponent)
+		}
+		componentInDevfile = localComponent.Name
+	}
```

Comment:
Do we want to print the local component/binding if --namespace is passed? I find it a little confusing to display the binding/component running in some other namespace to be displayed when I explicitly ask for some other namespace.

```sh
➜  102 odo list binding --namespace operators
 ✓  Listing ServiceBindings from the namespace "operators" [3ms]
 NAME                            APPLICATION                     SERVICES                                                 RUNNING IN 
 * my-nodejs-app-cluster-sample  my-nodejs-app-app (Deployment)  cluster-sample (Cluster.postgresql.k8s.enterprisedb.io)  None       
➜  102 odo list binding                      
 ✓  Listing ServiceBindings from the namespace "default" [9ms]
 NAME                            APPLICATION                     SERVICES                                                 RUNNING IN 
 * my-nodejs-app-cluster-sample  my-nodejs-app-app (Deployment)  cluster-sample (Cluster.postgresql.k8s.enterprisedb.io)  None       
```

Reply:
@kadel what's your opinion?

Reply:
If we decide to, I would prefer to make this change as part as another PR

Reply:
I agree it seems a bit confusing to display the local component/binding when requesting some other namespace (especially since the first line printed is `Listing {ServiceBindings,resources} from the namespace "$ns"`).
But as discussed, it is okay to change this in a separate PR.